src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,435 @@
+
+/**
+ * Gridfield component for nesting GridFields
+ */
+class GridFieldNestedForm extends AbstractGridFieldComponent implements
+    GridField_URLHandler,
+    GridField_ColumnProvider,
+    GridField_SaveHandler,
+    GridField_HTMLProvider,
+    GridField_DataManipulator
+{
+    use Configurable, GridFieldStateAware;
+    
+    const POST_KEY = 'GridFieldNestedForm';
+
+    private static $allowed_actions = [
+        'handleNestedItem'
+    ];
+
+    private static $max_nesting_level = 10;
+    
+    /**
+     * @var string
+     */
+    protected $name;
+    protected $expandNested = false;
+    protected $forceCloseNested = false;
+    protected $gridField = null;
+    protected $record = null;
+    protected $relationName = 'Children';
+    protected $inlineEditable = false;
+    protected $canExpandCheck = null;
+    protected $maxNestingLevel = null;
+    
+    public function __construct($name = 'NestedForm')
+    {
+        $this->name = $name;
+    }
+    
+    /**
+     * Get the grid field that this component is attached to
+     * @return GridField
+     */
+    public function getGridField()
+    {
+        return $this->gridField;
+    }
+
+    /**
+     * Get the relation name to use for the nested grid fields
+     * @return string
+     */
+    public function getRelationName()
+    {
+        return $this->relationName;
+    }
+    
+    /**
+     * Set the relation name to use for the nested grid fields
+     * @param string $relationName
+     */
+    public function setRelationName($relationName)
+    {
+        $this->relationName = $relationName;
+        return $this;
+    }
+
+    /**
+     * Get whether the nested grid fields should be inline editable
+     * @return boolean
+     */
+    public function getInlineEditable()
+    {
+        return $this->inlineEditable;
+    }
+
+    /**
+     * Set whether the nested grid fields should be inline editable
+     * @param boolean $editable
+     */
+    public function setInlineEditable($editable)
+    {
+        $this->inlineEditable = $editable;
+        return $this;
+    }
+    
+    /**
+     * Set whether the nested grid fields should be expanded by default
+     * @param boolean $expandNested
+     */
+    public function setExpandNested($expandNested)
+    {
+        $this->expandNested = $expandNested;
+        return $this;
+    }
+
+    /**
+     * Set whether the nested grid fields should be forced closed on load
+     * @param boolean $forceClosed
+     */
+    public function setForceClosedNested($forceClosed)
+    {
+        $this->forceCloseNested = $forceClosed;
+        return $this;
+    }
+
+    /**
+     * Set a callback to check which items in this grid that should show the expand link
+     * for nested gridfields. The callback should return a boolean value.
+     * @param callable $checkFunction
+     */
+    public function setCanExpandCheck($checkFunction)
+    {
+        $this->canExpandCheck = $checkFunction;
```

Comment:
```suggestion
    public function setCanExpandCallback($callback)
    {
        $this->canExpandCallback = $callback;
```
Just making it clearer by the naming what this actually is - the property holds a callback (or the name of a method).